faithcomesbyhearing/dbp
# FCBH-885 Plan/Playlists HLS Endpoint

# Description
- Updated timestamps endpoint
- Added v4 documentation to timestamps endpoint
- Added HLS to playlists
- Added support for full chapter items on playlists
- Removed Timestamp duplicate model file
- Change null logic on `StreamController` from `offset` to `timestamp`

## Issue Link
Original Story: [FCBH-885](https://fullstacklabs.atlassian.net/browse/FCBH-885) 
Review Story: [FCBH-1075](https://fullstacklabs.atlassian.net/browse/FCBH-1075) 

## How Do I QA This
- On your local environment run `php artisan migrate --database="dbp_users"` to run the migrations
- Run `https://dbp.test/open-api-v4.json` to get the new documentation
- Test `http://dbp.test/api/timestamps/{fileset_id}/{book_id}/{chapter}?v=4&key={KEY}` `GET` and verify that the endpoint is now retrieving the timestamps.
- Test `playlists/{id}/`, `/playlists?showDetails=true` endpoints and verify that now you get the path parameter with the hls file of the playlist
- Also, verify that now the playlist items have a new parameter `full_chapter`, this happens when the playlist item `verse_start` and `verse_end` are `null` 

**(*)Note:** If you want to reverse the migration run `php artisan migrate:rollback --database="dbp_users"`

## Checklist
